mainServiceController/src/main/java/org/venice/piazza/servicecontroller/messaging/handlers/ExecuteServiceHandler.java

```diff
@@ -117,125 +129,101 @@ public ExecuteServiceHandler(MongoAccessor accessor, CoreServiceProperties coreS
 	public ResponseEntity<String> handle (ExecuteServiceData data) {
 		ResponseEntity<String> responseEntity = null;
 		// Get the id from the data
-		String resourceId = data.getResourceId();
-		ResourceMetadata rMetadata = accessor.getResourceById(resourceId);
-		// Now get the mimeType for the request not using for now..
-		String requestMimeType = rMetadata.requestMimeType;
+		String serviceId = data.getServiceId();
+		Service sMetadata = accessor.getServiceById(serviceId);
+		// Default request mimeType application/json
+		String requestMimeType = "application/json";
```

Comment:
This should not be hard code but should be retrieved from the ExecuteServicePayload